File: Unity/Project Arena/Assets/Scripts/Maps/MapGenerator/Area.cs
namespace Maps.MapGenerator
{
    public class Area
    {
        public readonly int bottomRow;
        public readonly bool isCorridor;
        public readonly int leftColumn;
        public readonly int rightColumn;
        public readonly int topRow;

        public Area(
            int leftColumn,
            int bottomRow,
            int rightColumn,
            int topRow,
            bool isCorridor = false
        )
        {
            this.leftColumn = leftColumn;
            this.bottomRow = bottomRow;
            this.rightColumn = rightColumn;
            this.topRow = topRow;
            this.isCorridor = isCorridor;
        }
    }
}

File: Python/AIExperimentTools/internals/area.py

class Area:
    def __init__(self, left_column=0, bottom_row=0, right_column=0, top_row=0, is_corridor=False):
        self.bottomRow = bottom_row
        self.topRow = top_row
        self.leftColumn = left_column
        self.rightColumn = right_column
        self.isCorridor = is_corridor
        self.bottomRow = bottom_row

    def __eq__(self, other):
        if type(other) is type(self):
            return self.__dict__ == other.__dict__
        else:
            return False

    def __hash__(self):
        return hash(tuple(self.__dict__.items()))


def scale_area(area, scale):
    return Area(
        scale * area.leftColumn,
        scale * area.bottomRow,
        scale * area.rightColumn,
        scale * area.topRow,
        area.isCorridor,
    )
